rust/otap-dataflow/crates/otap/src/otap_batch_processor.rs

```diff
@@ -0,0 +1,1159 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! OTAP batch processor.
+//! Batches OtapPdata by count or timer; uses upstream OTAP batching for merge/split.
+
+use crate::OTAP_PROCESSOR_FACTORIES;
+use crate::pdata::OtapPdata;
+use async_trait::async_trait;
+use linkme::distributed_slice;
+use otap_df_config::error::Error as ConfigError;
+use otap_df_config::experimental::SignalType;
+use otap_df_config::node::NodeUserConfig;
+use otap_df_engine::config::ProcessorConfig;
+use otap_df_engine::error::Error as EngineError;
+use otap_df_engine::local::processor as local;
+use otap_df_engine::message::Message;
+use otap_df_engine::node::NodeId;
+use otap_df_engine::processor::ProcessorWrapper;
+use serde::Deserialize;
+use serde_json::Value;
+use std::num::NonZeroU64;
+use std::sync::Arc;
+// For optional conversion during flush/partitioning
+use otel_arrow_rust::otap::OtapArrowRecords;
+use otel_arrow_rust::otap::batching::make_output_batches;
+
+/// URN for the OTAP batch processor
+pub const OTAP_BATCH_PROCESSOR_URN: &str = "urn:otap:processor:batch";
+
+/// Default configuration values (parity-aligned as we confirm Go defaults)
+pub const DEFAULT_SEND_BATCH_SIZE: usize = 8192;
+/// Default upper bound on batch size used to chunk oversized inputs (in number of items).
+/// See Config::send_batch_max_size for normalization semantics.
+pub const DEFAULT_SEND_BATCH_MAX_SIZE: usize = 8192;
```

Comment:
Should the default be zero?
> send_batch_max_size (default = 0): The upper limit of the batch size. 0 means no upper limit of the batch size. This property ensures that larger batches are split into smaller units. It must be greater than or equal to send_batch_size.

https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor